### PR TITLE
Add learning rate to logs

### PR DESCRIPTION
**Pull request by epwalsh**
This adds the learning rate for all parameter groups to the logs for each epoch. This is especially helpful when trying to fine-tune the learning rate.

learning_rate_logging1

learning_rate_loggin2

### CODE REVIEW

**Comment by matt-gardner** [comment]
I really like including this in tensorboard; I'm less convinced that this is useful as a metric in the metrics table.  For one thing, we only show three decimal places there, and this could easily just be `0.000`, which would be confusing.

**Comment by epwalsh** [comment]
That's a good point. Well I could throw in a little check right here:
https://github.com/epwalsh/allennlp/blob/13c5ba63a42a9c8100cde3c94a6c45db05cb7bc1/allennlp/training/trainer.py#L653

like `if name.startswith("learning_rate"): continue` in order to avoid that

**Comment by matt-gardner** [comment]
Or maybe just do `self._tensorboard.add_train_scalar(...)` here: https://github.com/epwalsh/allennlp/blob/13c5ba63a42a9c8100cde3c94a6c45db05cb7bc1/allennlp/training/trainer.py#L533

Would that work?

**Comment by epwalsh** [comment]
If we do it that way then I think we would also need to put a line here: https://github.com/epwalsh/allennlp/blob/13c5ba63a42a9c8100cde3c94a6c45db05cb7bc1/allennlp/training/trainer.py#L757 with a call to `self._tensorboard.add_train_scalar` so that we get a per-epoch log, right?

**Comment by epwalsh** [comment]
@matt-gardner got it 👌, I only had per-epoch in mind. I just made that change. Let me know if you see anything else that I should update

File: allennlp/training/trainer.py [hunk 1]
```diff
@@ -747,6 +747,11 @@ def train(self) -> Dict[str, Any]:
             self._save_checkpoint(epoch, validation_metric_per_epoch, is_best=is_best_so_far)
             self._metrics_to_tensorboard(epoch, train_metrics, val_metrics=val_metrics)
             self._metrics_to_console(train_metrics, val_metrics)
+            for ind, param_group in enumerate(self._optimizer.param_groups):
```
Comment (matt-gardner) [inline allennlp/training/trainer.py]: `ind` is for `index`?  If so, just use `index` - for the price of just two characters, I don't have to guess what you mean ;).
